src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -924,9 +949,12 @@ private void WriteLocalFileHeaderAndDataIfNeeded()
                     WriteLocalFileHeader(false);
                     // we just want to copy the compressed bytes straight over, but the stream apis
                     // don't handle larger than 32 bit values, so we just wrap it in a stream
-                    using (MemoryStream compressedStream = new MemoryStream(_compressedBytes))
+                    foreach (Byte[] compressedBytes in _compressedBytes)
```

Comment:
The above comment doesn't make sense.  Stream.Write can work with any size array, which has a max length that's actually less than Int32.MaxValue.  We can address this in a follow-up, but there doesn't appear to be any reason here to wrap compressedBytes in a MemoryStream: it could just be ```_archive.ArchiveStream.Write(compressedBytes, 0, compressedBytes.Length);```.
cc: @FiveTimesTheFun